src/main/java/ca/on/oicr/pineryreports/reports/impl/LanesBillingReport.java

```diff
@@ -0,0 +1,446 @@
+      String before = cmd.getOptionValue(OPT_BEFORE.getLongOpt());
+      if (!before.matches(DATE_REGEX)) {
+        throw new ParseException("Before date must be in format yyyy-mm-dd");
+      }
+      this.end = before;
+    }
+  }
+
+  @Override
+  public String getTitle() {
+    return "Lanes Completion (Billing) "
+        + (start == null ? "Any Time" : start)
+        + " - "
+        + (end == null ? "Now" : end);
+  }
+  
+  private static class ReportObject {
+    private final List<DetailedObject> detailedData;
+    private final List<Map.Entry<String, Map<String, BigDecimal>>> summaryAsList;
+    
+    public ReportObject(List<DetailedObject> detailed, Map<String, Map<String, BigDecimal>> summary) {
+      detailed.sort(DetailedObject.detailedComparator);
+      this.detailedData = detailed;
+      this.summaryAsList = listifySummary(summary);
+    }
+    
+    public List<DetailedObject> getDetailedData() {
+      return detailedData;
+    }
+    
+    public List<Map.Entry<String, Map<String, BigDecimal>>> getSummaryAsList() {
+      return summaryAsList;
+    }
+  }
+  
+  private ReportObject completed;
+  private ReportObject failed;
+
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException {
+    // samples and instruments and instrument models will come in handy
+    Map<String, SampleDto> samplesById = mapSamplesById(pinery.getSample().all());
+    Map<Integer, InstrumentDto> instrumentsById = pinery.getInstrument().all().stream()
+        .collect(Collectors.toMap(InstrumentDto::getId, dto->dto));
+    Map<Integer, InstrumentModelDto> instrumentModelsById = pinery.getInstrumentModel().all().stream()
+        .collect(Collectors.toMap(InstrumentModelDto::getId, dto->dto));
+    // filter runs within the date range
+    Set<RunDto> newRuns = pinery.getSequencerRun().all().stream().filter(byEndedBetween(start, end)).collect(Collectors.toSet());
+    Set<RunDto> failedRuns = newRuns.stream().filter(run -> "Failed".equals(run.getState())).collect(Collectors.toSet());
+    newRuns.removeAll(failedRuns);
+    
+    completed = getReportAndSummaryData(newRuns, samplesById, instrumentsById, instrumentModelsById);
+    failed = getReportAndSummaryData(failedRuns, samplesById, instrumentsById, instrumentModelsById);
+  }
+  
+  private ReportObject getReportAndSummaryData(Collection<RunDto> runs, Map<String, SampleDto> samplesById,
+      Map<Integer, InstrumentDto> instrumentsById, Map<Integer, InstrumentModelDto> instrumentModelsById) {
+    List<DetailedObject> detailedData = new ArrayList<>();
+    Map<String, Map<String, BigDecimal>> summaryData = new TreeMap<>();
+    for (RunDto run : runs) {
+      String instrumentName = getInstrumentName(run.getInstrumentId(), instrumentsById);
+      String instrumentModel = getInstrumentModel(run.getInstrumentId(), instrumentsById, instrumentModelsById);
+      if (run.getPositions() == null) continue;; 
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This statement will not be executed conditionally; only the first statement will be. The rest will execute unconditionally. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2681)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)